// Generated by funapi_initiator
// This file is an example to illustrate how to use protobuf
// client-server messages on Funapi.
// You can edit this file to meet your taste.


// You should use a relative path to the system top-level directory.
// On Linux, it's /usr/include.
// To import /usr/include/funapi/network/fun_message.proto,
// use relative path like this:
import "funapi/network/fun_message.proto";

// Your specific message class.
message PbufEchoMessage {
  required string msg = 1;
}


message PbufAnotherMessage {
  optional string msg = 1;
}

message FacebookLoginMessage {
  required string facebook_uid = 1;
  required string facebook_access_token = 2;
}


extend FunMessage {
  optional PbufEchoMessage pbuf_echo = 16;
  optional PbufAnotherMessage pbuf_another = 17;
  optional FacebookLoginMessage facebook_login = 18;
}


import "funapi/service/multicast_message.proto";

message PbufHelloMessage
{
  optional string message = 1;
}
extend FunMulticastMessage
{
  optional PbufHelloMessage pbuf_hello = 16;
}
